draft-ietf-rtcweb-jsep.xml

```diff
@@ -1626,39 +1626,32 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
-          the ICE agent, which, if parsed successfully, will be added
+          <t>The addIceCandidate method provides an update to
+          the ICE agent via an IceCandidate object
+          <xref target="sec.ice-candidate-format"/>. If the
+          IceCandidate's candidate field is filled in, the IceCandidate
+          is treated as a new remote ICE candidate, which will be added
           to the current and/or pending remote description according to
-          the rules defined for Trickle ICE. The pair of MID and ufrag
-          is used to determine the m= section and ICE candidate
-          generation to which the candidate belongs. If the MID is not
-          present, the m= section index is used to look up the locally
-          generated MID (see
-          <xref target="sec.applying-a-remote-desc" />), which is used
-          in place of a supplied MID. If these values or the candidate
-          string are invalid, an error is generated.</t>
-
-          <t>The purpose of the ufrag is to resolve ambiguities when
-          trickle ICE is in progress during an ICE restart. If the
-          ufrag is absent, the candidate MUST be assumed to belong to
-          the most recently applied remote description. Connectivity
-          checks will be sent to the new candidate.</t>
-
-          <t>This method can also be used to provide an
-          end-of-candidates indication to the ICE agent, as defined in
-          <xref target="I-D.ietf-ice-trickle" />). The MID and ufrag
-          are used as described above to determine the m= section and
-          ICE generation for which candidate gathering is complete. If
-          the ufrag is not present, then the end-of-candidates
-          indication MUST be assumed to apply to the relevant m=
-          section in the most recently applied remote description. If
-          neither the MID nor the m= index is present, then the
-          indication MUST be assumed to apply to all m= sections in the
-          most recently applied remote description.</t>
-
-          <t>This call will result in a change to the state of the ICE
-          Agent, and may result in a change to media state if it
-          results in connectivity being established.</t>
+          the rules defined for Trickle ICE. Otherwise, the IceCandidate
+          is treated as an end-of-candidates indication, as defined in
+          <xref target="I-D.ietf-ice-trickle" />.</t>
+
+          <t>In either case, the m= section index, MID, and ufrag fields from
+          the supplied IceCandidate are used to determine which m= section
+          and ICE candidate generation the IceCandidate is for, as described
+          in <xref target="sec.ice-candidate-format" /> above. In the
+          specific case of an end-of-candidates indication, both the m= section
+          index and MID may be omitted to mean that the indication applies
+          to all m= sections in the specified ICE candidate generation.</t>
```

Comment:
You've switched from the receiver's perspective to the senders here. Perhaps
"If the object indicates end-of-candidates and both the m= section index and MID re omitted, that means that the candidate applies. Otherwise, if both are omitted, then the JSEP implementation MUST ignore the IceCandidate and return ann error"

Reply:
Agreed; rewrote to be clear this is from the receiver's perspective